doc/modules/core.rst

```diff
@@ -316,6 +257,78 @@ Once a :code:`SortingAnalyzer` object is saved to disk, it can be easily reloade
+    total_duration = sorting_analyzer.get_total_duration()
+    ### NOTE ###
+    # 'segment_index' is required for multi-segment objects
+    num_samples = sorting_analyzer.get_num_samples(segment_index=0)
+
+    # channel_ids and unit_ids
+    channel_ids = sorting_analyzer.channel_ids
+    unit_ids = sorting_analyzer.unit_ids
+
+To calculate extensions, we need to have included the module they come from. Most of the extensions live in the
+:code:`postprocessing` module (with the :code:`quality_metrics` extension
+in the :code:`qualitymetrics` module) , but there are some *core* extensions too:
+
+* :code:`random_spikes`: select random spikes for downstream analysis (e.g., extracting waveforms or fitting a PCA model)
+* :code:`templates`: estimate templates
+* :code:`waveforms`: extract waveforms for single spikes
+* :code:`noise_levels`: compute channel-wise noise levels
+
+Extensions have a parent/child structure. You can only compute a child _after_ you've computed the parent. For the core
+extensions, the structure is fairly straightforward. :code:`random_spikes` and :code:`noise_levels` have no parents. :code:`waveforms` is the child
+of :code:`random_spikes`. :code:`templates` is the child of :code:`waveforms` or :code:`random_spikes`, as it can be computed using either (albeit with
+different methods).
+
+.. note::
+
+    Some extension, like :code:`waveforms`, depend on other extension, like which spikes were randomly selected by :code:`random_spikes`.
+    So if we were to recalculate :code:`random_spikes`, the :code:`waveforms` will change (a little). To avoid this inconsistency,
+    spike interface deletes children if the parent is recalculated. E.g. if :code:`random_spikes` is recalculated, :code:`waveforms`
+    is deleted. This keeps consistency between your extensions, and is better for provenance.
+
+In practice, we use the :code:`compute` method to compute extensions. Provided the :code:`sorting_analyzer` is instantiated,
+additional extensions are computed as follows:
+
+.. code-block:: python
+
+    # compute some additional extensions
+    random_spikes_extension = sorting_analyzer.compute("random_spikes")
+    templates_extension = sorting_analyzer.compute("templates")
+    waveforms_extension = sorting_analyzer.compute("waveforms")
+
+    # each extension has a .data field: a dictionary with computed data
+    print(templates_extension.data.keys())
+
+.. code-block:: bash
+
+    >>> dict_keys(['average', 'std'])
+
+You can also pass parameters to the :code:`compute`, or compute several extensions at once. There are more details in
+the `PostProcessing docs <https://spikeinterface.readthedocs.io/en/latest/modules/postprocessing.html>`_.
+
+.. code-block:: python
+
+    # arguments can be passed directly to the compute function
+    # note that re-computing an extension will overwrite the existing one
+    waveform_extension_2 = sorting_analyzer.compute("waveforms", ms_before=2, ms_after=5)
```

Comment:
@h-mayorquin compute returns the estension of you pass a string. I agree that this could be confusing!

Reply:
I'm the opposite. I hate returning the extensions. I grab them when I need them with get_extension :) So my trick is I always use a list sorting_analyzer.compute(['waveforms']) and with the new option I can now pass params to this too ! (Thanks @alejoe91 !)

Reply:
Thanks for the different perspective @zm711.  Why do you dislike returning? You don't like assigning values?

@alejoe91 
I see thanks for the correction, I read the function and did not realize there is a return for the first case. My confusion arose because I was passing a list with a single element and I did not expect that to have a different behavior than passing only the string.

Reply:
Yeah. I don't often work with the extensions themselves currently. I do a lot of plotting and qcmetrics with SI so for me it's more about running those steps. Then when I have things done I pull only the extension I need data from. So I end up just polluting my variable namespace with poorly named scripting variables (definitely my problem, but self knowledge is important :)) So that's my main issue.

Reply:
Got you. Thanks.